### PR TITLE
Feat/pyinstaller

### PR DESCRIPTION
**Pull request by zzjc1234**
This pr modified canvas_app s.t. the code can be packed with pyinstaller

### CODE REVIEW

**Comment by linsyking** [comment]
Have you tested whether pyinstaller can work on this?

**Comment by zzjc1234** [comment]
Yes, the exe works well. I disablesd the output in console, I think users may not need it.

**Comment by zzjc1234** [comment]
The updator may have some unknown problem, but the updator.py should be modified anyway. Maybe I will fix it in another pr.

**Comment by linsyking** [comment]
For executables the old updater doesn't work because it is based on git

**Comment by linsyking** [comment]
Maybe we can just remove git update logic when building for executables.

**Comment by zzjc1234** [comment]
It seems that I didn't solve the dependency well, currently the exe only works under the dir with canvas_mgr.py and etc. I will try to fix it.

**Comment by linsyking** [comment]
Also we need to migrate some arguments from uvicorn to the executables, eg. the port and base_url

**Comment by linsyking** [comment]
I think we can remove the updater support. It is not very helpful and needs git dependency.

**Comment by linsyking** [comment]
do not merge until everything is set

**Comment by linsyking** [comment]
Please fix the build error.

**Comment by linsyking** [comment]
I think we should first merge it to the dev branch otherwise current users will be affected

File: start.py [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+            "handlers": ["file_handler"],
+            "level": "ERROR",
+        },
+    }
+
+    if path.exists("net_config.json"):
+        with open("net_config.json", "r", encoding="utf-8", errors="ignore") as f:
+            net_config = json.load(f)
+
+            usr_host = net_config["host"]
```
Comment (linsyking) [inline start.py:40]: If "host" is not a field in the json file, it will throw an  exception

File: start.py [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+            usr_port = net_config["port"]
+
+            if usr_host == "":
+                usr_host = "localhost"
+            if usr_port == "":
+                usr_port = 9283
```
Comment (linsyking) [inline start.py]: Do not write duplicate code

**Review by linsyking — CHANGES_REQUESTED** [review]

**Review by linsyking — APPROVED** [review]

File: start.py (hunk 2 above)
Comment (zzjc1234) [reply start.py]: fixed

**Pull request merged**